numpy/core/tests/test_dtype.py

```diff
@@ -151,11 +151,11 @@ def test_field_order_equality(self):
                       'formats': ['i4', 'f4'],
                       'offsets': [0, 4]})
         y = np.dtype({'names': ['B', 'A'],
-                      'formats': ['f4', 'i4'],
+                      'formats': ['i4', 'f4'],
                       'offsets': [4, 0]})
         assert_equal(x == y, False)
-        # But it is currently an equivalent cast:
-        assert np.can_cast(x, y, casting="equiv")
+        # This is an safe cast (not equiv) due to the different names:
```

Comment:
This is a really tricky example, as the way it was written, it described exactly the same bytes in memory, but just with a different order. With the change to use field order, I guess it makes sense that, as written, it would become "unsafe" (integer to float and vice versa). Might it be good to keep that as an example?

And then also insert the new example, where, because it is by field order, it is "safe".